src/charm.py

```diff
@@ -372,6 +398,77 @@ def _is_magmad_enabled(self) -> bool:
         )
         return not magma_service.returncode
 
+    def _install_configurations(self, event: OrchestratorAvailableEvent) -> bool:
+        """Install or update configuration files.
+
+        Returns:
+            True if any changes were applied
+        """
+        return any(
+            [
+                self._install_root_ca_certificate(event.root_ca_certificate),
+                self._install_agw_config(event),
+            ]
+        )
+
+    @staticmethod
+    def _install_root_ca_certificate(root_ca_certificate: bytes) -> bool:
+        """Install root ca certificate or update it.
+
+        Returns:
+            True if the root ca certificate was written
+        """
+        cert_dir = Path(CERT_DIR)
+        if not cert_dir.exists():
+            cert_dir.mkdir()
+        root_ca_file = cert_dir / "rootCA.pem"
+        if root_ca_file.exists():
+            if root_ca_file.read_bytes() == root_ca_certificate:
+                return False
+        root_ca_file.write_bytes(root_ca_certificate)
+        return True
+
+    def _install_agw_config(self, event: OrchestratorAvailableEvent) -> bool:
+        """Install AGW configuration file or update it.
+
+        Returns:
+            True if the configuration file was written
+        """
+        config = self._generate_config(event)
+        config_dir = Path(CONFIG_DIR)
+        if not config_dir.exists():
```

Comment:
As above.

Reply:
I replied above and will close this one, as I refactored to a single method.